File: solutions/python/anagram/2/anagram.py
"""Find anagrams of a target word from a list of candidate words."""

def find_anagrams(word, candidates):
    target_lower = word.lower()
    target_sorted = sorted(target_lower)
    return [
        candidate
        for candidate in candidates
        if candidate.lower() != target_lower and sorted(candidate.lower()) == target_sorted
    ]

File: solutions/javascript/anagram/1/anagram.js
export const findAnagrams = (target, candidates) => {
  const normalize = (word) => word.toLowerCase().split('').sort().join('');
  const targetNormalized = normalize(target);

  return candidates.filter(candidate => {
    // Not an anagram if same word (case-insensitive)
    if (candidate.toLowerCase() === target.toLowerCase()) {
      return false;
    }
    return normalize(candidate) === targetNormalized;
  });
};